source/common/quic/client_connection_factory_impl.cc

```diff
@@ -24,11 +24,21 @@ PersistentQuicInfoImpl::PersistentQuicInfoImpl(
           std::make_unique<quic::QuicCryptoClientConfig>(std::make_unique<EnvoyQuicProofVerifier>(
               stats_scope, getConfig(transport_socket_factory), time_source))) {}
 
+namespace {
+// TODO(mattklein123): This is mutable static info that is required for the QUICHE code. This was
+// preexisting but should be removed as we shouldn't be using global mutable static data.
```

Comment:
cc @danzh2010 @alyssawilk for comment on this one.

Reply:
you're welcome to TODO this my way, alternately we can probably stick both in PersistentQuicInfo, or make push promise index there and make quic_config_ const at which point it's legal I think?

Reply:
OK let me take another look.

Reply:
ah, and having checked in, bence is starting to remove server push upstream so hopefully push promise index will go away soonish regardless (plus remove a bunch of uncovered stubs)

Reply:
OK if this is going to change soon I will probably just punt since this is no different from what it was before.

Reply:
yeah, if you don't mind changing the TODO to the quic upstream bug that'd be great (since I'm tracking those).

Reply:
Why do we need to remove QuicClientConnectionFactoryImpl? It makes sense to remove the abstraction, but I think we need to stick some quiche related stuff in connection pool any way, why not just wrap everything in this factory, including config_? And this factory which depends on QUICHE can be hide behind ENVOY_ENABLE_QUIC

Reply:
> Why do we need to remove QuicClientConnectionFactoryImpl? 

Because the indirection (and lookup cost) isn't needed. Essentially, the factory is a global object, so what I have here is exactly the same as what we had before. My preference still is to just leave this as it is the same and we can sort it out in a future change? This PR is quite complicated and I'm trying to not complicate it further. :)

Reply:
A TODO to move config_ into PersistentQuicInfo is fine. Or even better TODO to move the creation logic into PersistentQuicInfo.

Reply:
OK will do.